client-tags/icon.md

```diff
@@ -0,0 +1,86 @@
+---
+title: `+icons` client tag
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "James Wheare"
+    period: 2016
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+icons` tag name. Instead, implementations SHOULD use the
+`+draft/icons` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification defines a client-only message tag to indicate message icons
+
+## Motivation
+
+To allow for richer display of messages with added visual context, a method for client and bot users to attach custom icons to messages is needed. This method is intended for message specific icons. For persistent avatars, a metadata key would be more appropriate.
+
+Visual clients are given a way to choose an appropriate icon resource for the display context, with a method for providing multiple icon sources, with different dimensions.
+
+A content type hint is also provided, to allow clients to filter unsupported image formats without downloading the resource.
+
+## Architecture
+
+### Dependencies
+
+Clients wishing to use this tag MUST negotiate the [`draft/message-tags`](../core/message-tags-3.3.html) capability with the server.
+
+### Format
+
+The icons tag is sent by a client with the client-only prefix `+` and its value is a JSON encoded array of image objects that reference icon resources for the message.
+
+    +draft/icons=<json>
+
+Image objects MAY contain any of the following members:
+
+| Member           | Description |
+| :--------------- | :---------- |
+| `src` (string)   | The Uniform Resource Indicator (URI) of an icon resource |
+| `sizes` (string) | A space separate list of pixel image dimensions in `wxh` format supported by the resource. Clients MAY use this to select an appropriate icon resource for the context and pixel density of the display |
+| `type` (string)  | A media type hint for the icon resource. Clients MAY use this to filter out unsupported icon types without downloading the resource |
+
+When an icon `src` is an HTTP resource, it SHOULD be referenced with an HTTPS protocol where available.
+
+When the icon is encoded as an [RFC2397 Data URI](https://tools.ietf.org/html/rfc2397) the length of the URI SHOULD be no more than 1024.
```

Comment:
SHOULD be no more than? Is perhaps SHOULD NOT exceed 1024 characters a better phrasing?